### PR TITLE
Various small fixes

### PR DESCRIPTION
**Pull request by godexsoft**
A few small imrpovements
- Fix a race condition
- Refactor RPCEngine to use a strategy for verifying user is admin or not
- Fix small bug in UT due to uninitialized field

### CODE REVIEW

File: src/backend/SimpleCache.cpp [hunk 1]
```diff
@@ -96,9 +96,9 @@ SimpleCache::getPredecessor(ripple::uint256 const& key, uint32_t seq) const
 std::optional<Blob>
 SimpleCache::get(ripple::uint256 const& key, uint32_t seq) const
 {
+    std::shared_lock lck{mtx_};
```
Comment (cindyyan317) [inline src/backend/SimpleCache.cpp:99]: Good catch~

File: src/main/main.cpp [hunk 2]
```diff
@@ -30,6 +30,7 @@
 #include <etl/ReportingETL.h>
 #include <log/Logger.h>
 #include <rpc/Counters.h>
+#include <rpc/RPCEngine.h>
```
Comment (cindyyan317) [inline src/main/main.cpp:33]: Why we add this header here?

File: src/webserver/HttpBase.h [hunk 3]
```diff
@@ -41,6 +41,7 @@
 #include <main/Build.h>
 #include <rpc/Counters.h>
 #include <rpc/RPC.h>
+#include <rpc/RPCEngine.h>
```
Comment (cindyyan317) [inline src/webserver/HttpBase.h:44]: it is not used in this header?

File: src/webserver/WsBase.h [hunk 4]
```diff
@@ -25,6 +25,7 @@
 #include <log/Logger.h>
 #include <rpc/Counters.h>
 #include <rpc/RPC.h>
+#include <rpc/RPCEngine.h>
```
Comment (cindyyan317) [inline src/webserver/WsBase.h:28]: it is not used in this header?

File: src/main/main.cpp (hunk 2 above)
Comment (godexsoft) [reply src/main/main.cpp:33]: For `RPCEngine::make_RPCEngine()` factory.

File: src/webserver/HttpBase.h (hunk 3 above)
Comment (godexsoft) [reply src/webserver/HttpBase.h:44]: `RPC::RPCEngine` is used. Could maybe forward declare but meh.

File: src/webserver/WsBase.h (hunk 4 above)
Comment (godexsoft) [reply src/webserver/WsBase.h:28]: Same as for HttpBase.h :)

**Review by cindyyan317 — APPROVED** [review]
Nice to catch that race condition!
All looks good. Please check some headers seem no need to be added.

**Review by cindyyan317 — APPROVED** [review]
Nice to catch that race condition!
All looks good. Please check some headers seem no need to be added.

**Pull request merged**